src/linear_algebra/matrix/nonsingular_inverse.lean

```diff
@@ -383,23 +383,101 @@ begin
+lemma inv_eq_nonsing_inv_of_invertible [invertible A] : ⅟ A = A⁻¹ :=
+begin
+  suffices : is_unit A,
+  { rw [←this.mul_left_inj, inv_of_mul_self, matrix.mul_eq_mul, nonsing_inv_mul],
+    rwa ←is_unit_iff_is_unit_det },
+  exact is_unit_of_invertible _
+end
+
+variables {A} {B}
+
+/-- "Lift" the proposition `is_unit A.det` to a constructive inverse of `A`. -/
+noncomputable def invertible_of_is_unit_det  (h : is_unit A.det) : invertible A :=
+⟨A⁻¹, nonsing_inv_mul A h, mul_nonsing_inv A h⟩
+
+/-- "Lift" the proposition `is_unit A` to a constructive inverse of `A`. -/
+noncomputable def invertible_of_is_unit  (h' : is_unit A) : invertible A :=
+by have h := (is_unit_iff_is_unit_det A).1 h';
+exact ⟨A⁻¹, nonsing_inv_mul A h, mul_nonsing_inv A h⟩
+
+lemma is_unit_det_of_left_inverse (h : B ⬝ A = 1) : is_unit A.det :=
 @is_unit_of_invertible _ _ _ (det_invertible_of_left_inverse _ _ h)
 
-lemma is_unit_det_of_right_inverse (B : matrix n n α) (h : A ⬝ B = 1) : is_unit A.det :=
+lemma is_unit_det_of_right_inverse (h : A ⬝ B = 1) : is_unit A.det :=
 @is_unit_of_invertible _ _ _ (det_invertible_of_right_inverse _ _ h)
 
-lemma nonsing_inv_left_right (B : matrix n n α) (h : A ⬝ B = 1) : B ⬝ A = 1 :=
+lemma nonsing_inv_left_right (h : A ⬝ B = 1) : B ⬝ A = 1 :=
 begin
-  have h' : is_unit B.det := B.is_unit_det_of_left_inverse A h,
+  have h' : is_unit B.det := is_unit_det_of_left_inverse h,
   calc B ⬝ A = (B ⬝ A) ⬝ (B ⬝ B⁻¹) : by simp only [h', matrix.mul_one, mul_nonsing_inv]
         ... = B ⬝ ((A ⬝ B) ⬝ B⁻¹) : by simp only [matrix.mul_assoc]
         ... = B ⬝ B⁻¹ : by simp only [h, matrix.one_mul]
         ... = 1 : mul_nonsing_inv B h',
 end
 
-lemma nonsing_inv_right_left (B : matrix n n α) (h : B ⬝ A = 1) : A ⬝ B = 1 :=
-B.nonsing_inv_left_right A h
+lemma nonsing_inv_right_left (h : B ⬝ A = 1) : A ⬝ B = 1 :=
+nonsing_inv_left_right h
+
+/-- If matrix A is left invertible, then its inverse equals its left inverse. -/
+lemma inv_eq_left_inv (h : B ⬝ A = 1) : A⁻¹ = B :=
+begin
+  have h1 :=  (is_unit_det_of_left_inverse h),
+  have h2 := matrix.invertible_of_is_unit_det h1,
+  have := @inv_of_eq_left_inv (matrix n n α) (infer_instance) A B h2 h,
+  simp* at *,
+end
+
+/-- If matrix A is right invertible, then its inverse equals its right inverse. -/
+lemma inv_eq_right_inv (h : A ⬝ B = 1) : A⁻¹ = B :=
+begin
+  have h1 :=  (is_unit_det_of_right_inverse h),
+  have h2 := matrix.invertible_of_is_unit_det h1,
+  have := @inv_of_eq_right_inv (matrix n n α) (infer_instance) A B h2 h,
+  simp* at *,
+end
+
+/-- We can construct an instance of invertible A if A has a left inverse. -/
+noncomputable
+def invertible_of_left_inverse (h: B ⬝ A = 1) : invertible A :=
+matrix.invertible_of_is_unit_det (is_unit_det_of_left_inverse h)
+
+/-- We can construct an instance of invertible A if A has a right inverse. -/
+noncomputable
+def invertible_of_right_inverse (h: A ⬝ B = 1) : invertible A :=
+matrix.invertible_of_is_unit_det (is_unit_det_of_right_inverse h)
```

Comment:
You should be able to do this computably - you know the inverse is `B`.

Reply:
Eric, really good suggestion!